src/server/acl/validator.cc

```diff
@@ -86,6 +97,45 @@ namespace dfly::acl {
   return {keys_allowed, AclLog::Reason::KEY};
 }
 
+[[nodiscard]] std::pair<bool, AclLog::Reason> IsUserAllowedToInvokePubSubCommand(
+    bool literal_match, const std::vector<uint64_t>& acl_commands, const AclPubSub& pub_sub,
+    CmdArgList tail_args, const CommandId& id) {
+  const size_t index = id.GetFamily();
+  const uint64_t command_mask = id.GetBitIndex();
+  DCHECK_LT(index, acl_commands.size());
+
+  const bool command = (acl_commands[index] & command_mask) != 0;
+
+  if (!command) {
+    return {false, AclLog::Reason::COMMAND};
+  }
```

Comment:
that's code from above, I'm sure we can move it to the common function 🙂

Reply:
wanted to refactor this as well, good catch +1